### PR TITLE
Support animated stickers on Android

### PR DESCRIPTION
**Pull request by simonzhexu**
Animated Stickers is coming.

A list of things we are supporting in this pull request:
1. Support exporting animated stickers to WhatsApp
2. Tap to preview animated stickers.
3. Animated sticker pack indicator
4. A sample animated sticker pack asset.

### CODE REVIEW

**Review by chenhan0206 — APPROVED** [review]

**Comment by MarsVard** [comment]
Are you missing the boolean indicating the sticker itself is animated or not by accident or did that just get deprecated in the ContentProvider?

**Comment by simonzhexu** [comment]
> Are you missing the boolean indicating the sticker itself is animated or not by accident or did that just get deprecated in the ContentProvider? Ps. When can we expect a beta to test this on?

@MarsVard , there is a boolean in the ContentProvider that indicates whether a pack is animated or not. There is no individual boolean for stickers.

File: Android/app/src/main/java/com/example/samplestickerapp/StickerPackDetailsActivity.java [hunk 1]
```diff
@@ -43,6 +46,8 @@
     public static final String EXTRA_SHOW_UP_BUTTON = "show_up_button";
     public static final String EXTRA_STICKER_PACK_DATA = "sticker_pack";
 
+
+
```
Comment (chenhan0206) [inline Android/app/src/main/java/com/example/samplestickerapp/StickerPackDetailsActivity.java]: remove？

File: Android/app/src/main/java/com/example/samplestickerapp/StickerPackDetailsActivity.java [hunk 2]
```diff
@@ -51,8 +56,11 @@
     private View alreadyAddedText;
     private StickerPack stickerPack;
     private View divider;
+    private SimpleDraweeView expandedStickerView;
     private WhiteListCheckAsyncTask whiteListCheckAsyncTask;
 
+
+
```
Comment (chenhan0206) [inline Android/app/src/main/java/com/example/samplestickerapp/StickerPackDetailsActivity.java]: ditto

File: Android/app/src/main/java/com/example/samplestickerapp/ContentFileParser.java [hunk 3]
```diff
@@ -165,7 +169,9 @@ private static StickerPack readStickerPack(@NonNull JsonReader reader) throws IO
                         }
                     }
                     reader.endArray();
-                } else {
+
+                }
+                else {
```
Comment (chenhan0206) [inline Android/app/src/main/java/com/example/samplestickerapp/ContentFileParser.java]: revert？

Comment (simonzhexu) [reply Android/app/src/main/java/com/example/samplestickerapp/ContentFileParser.java]: yeah, this one is solved in the last commit, reformat.

File: Android/app/src/main/java/com/example/samplestickerapp/StickerPackValidator.java [hunk 4]
```diff
@@ -126,14 +128,17 @@ private static void validateSticker(@NonNull Context context, @NonNull final Str
...
             throw new IllegalStateException("no file path for sticker, sticker pack identifier:" + identifier);
         }
-        validateStickerFile(context, identifier, sticker.imageFileName);
+        validateStickerFile(context, identifier, sticker.imageFileName, animatedStickerPack);
     }
 
-    private static void validateStickerFile(@NonNull Context context, @NonNull String identifier, @NonNull final String fileName) throws IllegalStateException {
+    private static void validateStickerFile(@NonNull Context context, @NonNull String identifier, @NonNull final String fileName, final boolean animatedStickerPack) throws IllegalStateException {
         try {
             final byte[] bytes = StickerPackLoader.fetchStickerAsset(identifier, fileName, context.getContentResolver());
```
Comment (chenhan0206) [inline Android/app/src/main/java/com/example/samplestickerapp/StickerPackValidator.java:137]: better name to something like stickerInBytes

File: Android/app/src/main/java/com/example/samplestickerapp/StickerPackValidator.java [hunk 5]
```diff
@@ -26,15 +26,17 @@
 import java.util.List;
 
 class StickerPackValidator {
-    private static final int STICKER_FILE_SIZE_LIMIT_KB = 100;
+    private static final int STATIC_STICKER_FILE_LIMIT_KB = 1000;
+    private static final int ANIMATED_STICKER_FILE_LIMIT_KB = 3000;
```
Comment (chenhan0206) [inline Android/app/src/main/java/com/example/samplestickerapp/StickerPackValidator.java]: Is this the right file size limit? It is 100kb and 500 kb according to the spec?

**Comment by pratikbutani** [comment]
Thank you so much 🥇

**Pull request merged**